src/Interpreters/DDLWorker.cpp

```diff
@@ -1046,6 +1051,13 @@ String DDLWorker::enqueueQuery(DDLLogEntry & entry)
     zookeeper->createAncestors(query_path_prefix);
 
     String node_path = zookeeper->create(query_path_prefix, entry.toString(), zkutil::CreateMode::PersistentSequential);
+    if (max_pushed_entry_metric) {
+        DB::ReadBufferFromString in(node_path.substr(query_path_prefix.length()));
```

Comment:
`ReadBufferFromString` works with raw data from passed string, we cannot pass temporary string here, it will cause use after free error.number